packages/typescript-estree/src/node-utils.ts

```diff
@@ -611,7 +611,7 @@ export function convertToken(
   } else {
     return {
       type: tokenType,
-      value,
+      value: value as never,
```

Comment:
According to [`no-explicit-any`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-explicit-any.md), I should use `never` instead of `any`

Reply:
you should be able to do `value as Token['value']`
prefer that or use an `@ts-expect-error` with a description.

`as never` / `as any` are generally just hacky and make it easy to forget WHY you did it

Reply:
`value as TSESTree.Token['value']` isn't working unfortunately 😕

<img width="724" alt="" src="https://user-images.githubusercontent.com/6643991/121809951-2c2c6b00-cc5f-11eb-867b-998a6fedd851.png">